raiden/network/transport/matrix.py

```diff
@@ -536,22 +541,61 @@ def _network_name(self) -> str:
     def _private_rooms(self) -> bool:
         return bool(self._config.get('private_rooms'))
 
-    def _login_or_register(self):
+    def _login_or_register(self, prev_user_id=None, prev_access_token=None, prev_sync_token=None):
+        base_username = to_normalized_address(self._raiden_service.address)
+        _match_user = re.match(
+            f'^@{re.escape(base_username)}.*:{re.escape(self._server_name)}$',
+            prev_user_id or '',
+        )
+        if _match_user:  # same user as before
+            self.log.debug('Trying previous user login', user_id=prev_user_id)
+            self._client.user_id = prev_user_id
+            self._client.token = self._client.api.token = prev_access_token
+            self._client.hs = self._server_name
```

Comment:
The interface supports setting it through constructor, but as we're reusing the client's instance, I did the same it'd do. Again, being economic in the number of required changes, specially avoiding libs ones

Reply:
Nonetheless, I don't think it makes sense to `login_with_token`, as it'd not perform any actual query or http request, and is just about setting data which is already there, and part of the public interface of the client. I'll check on how to reinstantiate client on start, instead of construction.

Reply:
Done with raiden-network/raiden-libs#153 's `set_access_token`